### PR TITLE
Supplanting review bot

### PR DESCRIPTION
**Pull request by ben-pace**
So I will explain what this PR does:

* It re-adds the `<PostsAnnualReviewMarketTag>` Component to `<FooterTagList>` (which renders it at the top of post pages), and fixes it up to no longer refer to a comment but just to have a standard hover-over
* It adds a new field to the schema/table `ManifoldProbabilitiesCaches` for tracking the URL of the associated Manifold Market
* It deletes the field `annualReviewMarketCommentId` from the posts Schema, and deletes all the code that created Review Bot comments on posts

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208307480599924) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/tagging/FooterTagList.tsx [hunk 1]
```diff
@@ -313,6 +313,13 @@ const FooterTagList = ({
       )
     )
 
+  // const commentString = `<p>The <a href="https://www.lesswrong.com/bestoflesswrong">LessWrong Review</a> runs every year to select the posts that have most stood the test of time. This post is not yet eligible for review, but will be at the end of ${2023+1}. The top fifty or so posts are featured prominently on the site throughout the year.</p><p>Hopefully, the review is better than karma at judging enduring value. If we have accurate prediction markets on the review results, maybe we can have better incentives on LessWrong today. <a href="${marketUrl}">Will this post make the top fifty?</a></p>
```
Comment (Raemon) [inline packages/lesswrong/components/tagging/FooterTagList.tsx:316]: remove

File: packages/lesswrong/lib/collections/posts/schema.tsx [hunk 2]
```diff
@@ -795,21 +796,65 @@ const schema: SchemaType<"Posts"> = {
...
 
-  annualReviewMarketCommentId: {
-    ...foreignKeyField({
-      idFieldName: 'annualReviewMarketCommentId',
-      resolverName: 'annualReviewMarketComment',
-      collectionName: 'Comments',
-      type: 'Comment',
-      nullable: true
-    }),
+  // // This is deprecated, and if you're reading this in the future, delete it
```
Comment (Raemon) [inline packages/lesswrong/lib/collections/posts/schema.tsx:799]:
```suggestion
  // // This is deprecated, and if you're reading this in the future, you can probably delete it
```

File: packages/lesswrong/lib/collections/posts/schema.tsx [hunk 3]
```diff
@@ -795,21 +796,65 @@ const schema: SchemaType<"Posts"> = {
...
+    type: Number,
     optional: true,
     nullable: true,
     canRead: ['guests'],
-    canCreate: ['admins'],
-    canUpdate: ['admins'],
-    hidden: !isLWorAF,
-    group: formGroups.adminOptions,
+    hidden: !isLWorAF
+    // Implementation in postResolvers.ts
```
Comment (Raemon) [inline packages/lesswrong/lib/collections/posts/schema.tsx:823]:
```suggestion
    // Implementation in postResolvers.ts TODO: migrate this and other annual review market resolvers into the schema, for nicer developer experience
```

File: packages/lesswrong/server/callbacks/votingCallbacks.ts [hunk 4]
```diff
@@ -264,36 +250,7 @@ async function maybeCreateReviewMarket({newDocument, vote}: VoteDocTuple, collec
...
-        data: commentString
-      }}
-    },
-    currentUser: botUser,
-    context: createAdminContext()
-  })
-
-
-  return result.data
+  Posts.rawUpdateOne(post._id, {$set: {manifoldReviewMarketId: liteMarket.id}})
```
Comment (Raemon) [inline packages/lesswrong/server/callbacks/votingCallbacks.ts:253]:
```suggestion
  void Posts.rawUpdateOne(post._id, {$set: {manifoldReviewMarketId: liteMarket.id}})
```

File: packages/lesswrong/server/migrations/20240913T193135.add_url_to_manifoldProbabilitiesCache.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  const context = createAdminContext();
+
+  const filteredUpdateMarketInfo = filterNonnull(updatedMarketInfo.map(({ marketId, marketInfo }) => 
+    marketInfo && marketId ? { marketId, marketInfo } : null
+  ));
+
+  if (filteredUpdateMarketInfo.length === updatedMarketInfo.length) {
+    console.log("All markets have a url, proceeding with the update");
+  } else {
+    console.log("Some markets do not have a url, aborting");
```
Comment (Raemon) [inline packages/lesswrong/server/migrations/20240913T193135.add_url_to_manifoldProbabilitiesCache.ts:66]:
```suggestion
    // eslint-disable-next-line no-console
    console.log("Some markets do not have a url, aborting");
```

File: packages/lesswrong/components/posts/PostsAnnualReviewMarketTag.tsx [hunk 6]
```diff
@@ -26,35 +68,34 @@ const PostsAnnualReviewMarketTag = ({ post, annualReviewMarketInfo, classes }: {
...
 
-  const { CommentsNode, LWPopper } = Components;
+  const { HoverOver, ContentStyles, ContentItemBody } = Components;
   const { anchorEl, hover, eventHandlers } = useHover();
 
-  const { annualReviewMarketComment } = post;
+  const year = annualReviewMarketInfo.year
+  const marketUrl = annualReviewMarketInfo.url
+
+  const golden = (highlightMarket(annualReviewMarketInfo)) ? "expectedWinner" : "expectedLoser"
```
Comment (b0b3rt) [inline packages/lesswrong/components/posts/PostsAnnualReviewMarketTag.tsx:77]: I'd probably call this `marketOutcomeClass`, or something more representative of what it's doing

File: packages/lesswrong/components/posts/PostsPage/LWPostsPageHeader.tsx [hunk 7]
```diff
@@ -6,6 +6,7 @@ import classNames from 'classnames';
 import { getHostname, getProtocol } from './PostsPagePostHeader';
 import { postGetLink, postGetLinkTarget } from '@/lib/collections/posts/helpers';
 import { BOOKUI_LINKPOST_WORDCOUNT_THRESHOLD } from './PostBodyPrefix';
+import { AnnualReviewMarketInfo } from '@/lib/collections/posts/annualReviewMarkets';
```
Comment (b0b3rt) [inline packages/lesswrong/components/posts/PostsPage/LWPostsPageHeader.tsx:9]: `import type`

File: packages/lesswrong/components/posts/PostsPage/LWPostsPageHeaderTopRight.tsx [hunk 8]
```diff
@@ -3,6 +3,7 @@ import { Components, registerComponent } from '../../../lib/vulcan-lib';
 import { AnalyticsContext } from '@/lib/analyticsEvents';
 import { getVotingSystemByName } from '@/lib/voting/votingSystems';
 import classNames from 'classnames';
+import { AnnualReviewMarketInfo } from '@/lib/collections/posts/annualReviewMarkets';
```
Comment (b0b3rt) [inline packages/lesswrong/components/posts/PostsPage/LWPostsPageHeaderTopRight.tsx:6]: `import type`

File: packages/lesswrong/components/tagging/FooterTagList.tsx [hunk 9]
```diff
@@ -354,8 +354,9 @@ const FooterTagList = ({
           )
       )}
       {!hidePostTypeTag && postType}
+      {/* {post.manifoldReviewMarketId &&reviewMarketTag} */}
```
Comment (b0b3rt) [inline packages/lesswrong/components/tagging/FooterTagList.tsx:357]: cleanup

File: packages/lesswrong/lib/annualReviewMarkets.ts [hunk 10]
```diff
@@ -4,16 +4,19 @@ export type AnnualReviewMarketInfo = {
...
 export const getMarketInfo = (post: PostsBase): AnnualReviewMarketInfo | undefined => {
   if (typeof post.annualReviewMarketProbability !== 'number') return undefined
   if (typeof post.annualReviewMarketIsResolved !== 'boolean') return undefined
   if (typeof post.annualReviewMarketYear !== 'number') return undefined
+  if (typeof post.annualReviewMarketUrl !== 'string') return undefined
   return {
     probability: post.annualReviewMarketProbability,
     isResolved: post.annualReviewMarketIsResolved,
-    year: post.annualReviewMarketYear
+    year: post.annualReviewMarketYear,
```
Comment (b0b3rt) [inline packages/lesswrong/lib/annualReviewMarkets.ts:18]: Shouldn't this entire file be deleted, if everything's been moved to `/collections/posts/annualReviewMarkets.ts`?

File: packages/lesswrong/lib/collections/posts/schema.tsx [hunk 11]
```diff
@@ -795,21 +796,65 @@ const schema: SchemaType<"Posts"> = {
...
 
-  annualReviewMarketCommentId: {
-    ...foreignKeyField({
-      idFieldName: 'annualReviewMarketCommentId',
-      resolverName: 'annualReviewMarketComment',
-      collectionName: 'Comments',
-      type: 'Comment',
-      nullable: true
-    }),
+  // // This is deprecated, and if you're reading this in the future, you can probably delete it
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/posts/schema.tsx:799]: Any reason not to delete it now?

File: packages/lesswrong/server/callbacks/votingCallbacks.ts [hunk 12]
```diff
@@ -264,36 +250,7 @@ async function maybeCreateReviewMarket({newDocument, vote}: VoteDocTuple, collec
...
-        data: commentString
-      }}
-    },
-    currentUser: botUser,
-    context: createAdminContext()
-  })
-
-
-  return result.data
+  void Posts.rawUpdateOne(post._id, {$set: {manifoldReviewMarketId: liteMarket.id}})
```
Comment (b0b3rt) [inline packages/lesswrong/server/callbacks/votingCallbacks.ts:253]: I don't know if it's important for this to be `await`ed, but the previous code was, so unless you're sure this is safe, I'd still use `await`

File: packages/lesswrong/server/callbacks/votingCallbacks.ts [hunk 13]
```diff
@@ -15,7 +15,7 @@ import { createAdminContext } from '../vulcan-lib/query';
 import Tags from '../../lib/collections/tags/collection';
```
Comment (oetherington) [inline packages/lesswrong/server/callbacks/votingCallbacks.ts:15]: Github won't let me leave a comment on the correct line, but the `Comments` import on line 13 is now unused

**Review by oetherington — COMMENTED** [review]
No objections from EAF (the failing test should be fixed by merging master)

File: packages/lesswrong/server/callbacks/votingCallbacks.ts (hunk 12 above)
Comment (Raemon) [reply packages/lesswrong/server/callbacks/votingCallbacks.ts:253]: Previously this was doing a different thing where it returned a ReviewBot comment, which (I think) we did something with, and now the last awaitable thing is doing an entirely different thing that we aren't using for anything else

**Review by b0b3rt — APPROVED** [review]

**Pull request merged**